packages/fxa-content-server/app/scripts/lib/payment-server.js

```diff
@@ -35,31 +39,51 @@ const PaymentServer = {
       flowId,
     } = view.metrics.getFlowEventMetadata();
 
-    return view
-      .getSignedInAccount()
-      .createOAuthToken(managementClientId, {
-        scope: managementScopes,
-        ttl: managementTokenTTL,
-      })
-      .then((accessToken) => {
-        const queryString = Url.objToSearchString({
-          // device_id, flow_begin_time, and flow_id need to be propagated to
-          // the payments server so that the user funnel can be traced from the RP,
-          // through the content server, and to the payments server, appearing as
-          // the same user throughout.
-          device_id: deviceId,
-          flow_begin_time: flowBeginTime,
-          flow_id: flowId,
-          ...queryParams,
-        });
+    const queryString = Url.objToSearchString({
+      // device_id, flow_begin_time, and flow_id need to be propagated to
+      // the payments server so that the user funnel can be traced from the RP,
+      // through the content server, and to the payments server, appearing as
+      // the same user throughout.
+      device_id: deviceId,
+      flow_begin_time: flowBeginTime,
+      flow_id: flowId,
+      ...queryParams,
+    });
+    const url = `${managementUrl}/${redirectPath}${queryString}`;
+    const account = view.getSignedInAccount();
+    const unauthenticatedRedirect = () => {
+      if (isAllowedUnauthenticatedRoute(redirectPath)) {
+        return view.navigateAway(url);
+      }
+
+      view.navigate('/');
+    };
+
+    if (account) {
+      return account
```

Comment:
I think using async functions might allow us to read the code a little easier.

Reply:
heh I fought my urge to use async/await here to be consistent with the rest of the content server frontend.  Yes, happy to make that change.  👍